lib/bamboo/test.ex

```diff
@@ -90,23 +94,91 @@ defmodule Bamboo.Test do
       assert_delivered_email(unsent_email) # Will fail
   """
   def assert_delivered_email(%Bamboo.Email{} = email) do
-    import ExUnit.Assertions
     email = Bamboo.Mailer.normalize_addresses(email)
-    assert_received {:delivered_email, ^email}
+
+    do_assert_delivered_email(email)
+  end
+
+  defp do_assert_delivered_email(email) do
+    receive do
+      {:delivered_email, ^email} -> true
+    after
+      @timeout -> flunk_with_email_list(email)
+    end
+  end
+
+  defp flunk_with_email_list(email) do
+    if Enum.empty?(delivered_emails) do
+      flunk """
+      There were 0 emails delivered to this process.
+
+      If you expected an email to be sent, try these ideas:
+
+        1) Make sure you call deliver_now/1 or deliver_later/1 to deliver the email
+        2) Make sure you are using the Bamboo.TestAdapter
+        3) Use the process name feature of Bamboo.Test. This will allow Bamboo.Test
+           to work across processes: use Bamboo.Test, process_name: :my_test_name
+        4) If you are writing an acceptance test through a headless browser, see
```

Comment:
Yeah it's kind of iffy, because you can't use the async tests with them. So I'm trying to let people know that. Maybe I could rephrase it a bit?

```
If this is an acceptance test that is run through a headless browser, you can instead write a controller test, or set the process_name option as described in option 3.
```

I feel like that gives people the 2 options. They can either switch to a different type of test or set the process name option. So it doesn't really say which way is best, but does give you ways to handle it. What do you think?